samples/snippets/src/test/java/pubsublite/beam/BeamReadWriteIT.java

```diff
@@ -0,0 +1,189 @@
+    assertNotNull(
+        "Environment variable " + varName + " is required to perform these tests.",
+        System.getenv(varName));
+  }
+
+  @Rule public Timeout globalTimeout = Timeout.seconds(300); // 5 minute timeout
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_CLOUD_PROJECT_NUMBER");
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+
+    // Create a topic.
+    CreateTopicExample.createTopicExample(cloudRegion, zoneId, projectNumber, topicId, partitions);
+    assertThat(bout.toString()).contains("created successfully");
+
+    // Create a subscription.
+    CreateSubscriptionExample.createSubscriptionExample(
+        cloudRegion, zoneId, projectNumber, topicId, subscriptionId);
+    assertThat(bout.toString()).contains("created successfully");
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    // Delete a topic.
+    DeleteTopicExample.deleteTopicExample(cloudRegion, zoneId, projectNumber, topicId);
+    assertThat(bout.toString()).contains("deleted successfully");
+
+    // Delete a subscription.
+    DeleteSubscriptionExample.deleteSubscriptionExample(
+        cloudRegion, zoneId, projectNumber, subscriptionId);
+    assertThat(bout.toString()).contains("deleted successfully");
+
+    System.setOut(null);
+  }
+
+  // This static out of band communication is needed to retain serializability.
+  @GuardedBy("this")
+  private static final List<SequencedMessage> testQuickstartReceived = new ArrayList<>();
+
+  private static synchronized void addMessageTestQuickstartReceived(SequencedMessage message) {
+    testQuickstartReceived.add(message);
+  }
+
+  private static synchronized List<SequencedMessage> getTestQuickstartReceived() {
+    return ImmutableList.copyOf(testQuickstartReceived);
+  }
+
+  private static PTransform<PCollection<? extends SequencedMessage>, PCollection<Void>>
+      collectTestQuickstart() {
+    return MapElements.via(
+        new SimpleFunction<SequencedMessage, Void>() {
+          @Override
+          public Void apply(SequencedMessage input) {
+            BeamReadWriteIT.addMessageTestQuickstartReceived(input);
+            return null;
+          }
+        });
+  }
+
+  @Test
+  public void testQuickstart() throws Exception {
+    StreamingOptions options = PipelineOptionsFactory.as(StreamingOptions.class);
+    options.setRunner(DirectRunner.class);
+    options.setStreaming(false);
+    options.as(DirectOptions.class).setBlockOnRun(false);
+    // Publish some messages
+    {
+      Pipeline p = Pipeline.create(options);
+      WriteMessagesExample.writeMessagesExample(
+          cloudRegion, zoneId, projectNumber, topicId, messageCount, p);
+      // Message writing is bounded.
+      assertThat(p.run().waitUntilFinish(Duration.standardMinutes(2)))
+          .isEqualTo(PipelineResult.State.DONE);
+    }
```

Comment:
Checkstyle wants level 6 indentation here. 

https://github.com/googleapis/java-pubsublite/pull/686/checks?check_run_id=2829596976

Reply:
I think checkstyle doesn't know how to handle scope blocks. The suggestion is unreasonable: it would look so strange for the block curly braces to be indented an extra time.

This passes lint, so I'm going to nack this particular suggestion.